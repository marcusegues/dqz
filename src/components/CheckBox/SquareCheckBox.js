// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../styles/Scaling';

export const SquareCheckBox = ({ checked }: { checked: boolean }) => (
  <View style={{ flexDirection: 'row', alignItems: 'center' }}>
    {checked ? (
      <MaterialCommunityIcons
        name="checkbox-marked"
        size={moderateScale(25)}
        color="#4A4A4A"
      />
    ) : (
      <MaterialCommunityIcons
        name="checkbox-blank-outline"
        size={moderateScale(25)}
        color="#4A4A4A"
      />
    )}
  </View>
);
