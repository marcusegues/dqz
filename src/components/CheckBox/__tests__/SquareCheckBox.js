import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Touchable from 'react-native-platform-touchable';
import { SquareCheckBox } from '../SquareCheckBox';

describe('SquareCheckBox', () => {
  test('renders according to snapshot unchecked', () => {
    const component = renderer
      .create(<SquareCheckBox checked={false} onPress={() => {}} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test('renders according to snapshot checked', () => {
    const component = renderer
      .create(<SquareCheckBox checked onPress={() => {}} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  test('we can click it', () => {
    const clickFn = jest.fn();
    const component = shallow(<SquareCheckBox onPress={clickFn} />);
    component.find(Touchable).simulate('press');
    expect(clickFn).toBeCalled();
  });
});
