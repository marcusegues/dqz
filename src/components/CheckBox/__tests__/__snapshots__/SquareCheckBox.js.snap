// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareCheckBox renders according to snapshot checked 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#4A4A4A",
            "fontSize": 39.285714285714285,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï„²
    </Text>
  </View>
</View>
`;

exports[`SquareCheckBox renders according to snapshot unchecked 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
